### DISCUSSION TITLE
How to mock/test auth.uid()

### DISCUSSION THREAD

**Discussion by benjamin-kirkbride** [supabase/supabase#4799] [Questions] [answered]
https://github.com/supabase/supabase/issues/4794 includes the snippet:

```sql
begin;
set local "request.jwt.claims" to '{ "sub": "d0d4b028-2f87-523b-a580-0f3e85e6ff33", "email": "my@email.com" }';
select auth.uid();
commit;
```

and it does seem to work as long as you `SET ROLE authenticated` and stay in a single transaction (the caveat as described by that issue).

Is this the recommended way of doing this? Or is there a better way?

**Comment by steve-chavez** [comment, marked as answer]
Yes, that's good. I also use the same idea for [doing tests with pgTAP](https://github.com/steve-chavez/socnet/tree/master/tests).

> and it does seem to work as long as you SET ROLE authenticated 

Note that you can also change roles with `SET LOCAL ROLE authenticated`, as shown [here](https://github.com/steve-chavez/socnet/blob/master/tests/anons.sql#L7).

**Reply by benjamin-kirkbride** [reply]
> The effects of SET LOCAL last only till the end of the current transaction, whether committed or not.

Awesome, thanks!
